master/internal/resourcemanagers/priority_test.go

```diff
@@ -0,0 +1,271 @@
+
+	agents := make([]*mockAgent, 0)
+	groups := []*mockGroup{
+		{id: "group1", priority: &lowerPriority},
+		{id: "group2", priority: &higherPriority},
+	}
+	tasks := []*mockTask{
+		{id: "task1", slotsNeeded: 4, group: groups[0]},
+		{id: "task2", slotsNeeded: 1, group: groups[0]},
+		{id: "task3", slotsNeeded: 0, group: groups[1]},
+		{id: "task4", slotsNeeded: 0, group: groups[1]},
+		{id: "task5", slotsNeeded: 4, group: groups[1]},
+	}
+
+	system := actor.NewSystem(t.Name())
+	taskList, mockGroups, _ := setupSchedulerStates(t, system, tasks, groups, agents)
+
+	pendingTasksByPriority, _ := sortTasksByPriorityAndTimestamp(taskList, mockGroups, "")
+	for priority, tasksInPriority := range pendingTasksByPriority {
+		switch priority {
+		case lowerPriority:
+			expectedTasks := []*mockTask{tasks[0], tasks[1]}
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		case higherPriority:
+			expectedTasks := []*mockTask{tasks[4]}
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		default:
+			panic("unexpected priority")
+		}
+	}
+
+	task5, _ := taskList.GetTaskByID("task5")
+	taskList.SetAllocations(
+		task5.TaskActor,
+		&ResourcesAllocated{
+			task5.ID,
+			"",
+			[]Allocation{containerAllocation{}},
+		},
+	)
+
+	_, scheduledTasksByPriority := sortTasksByPriorityAndTimestamp(taskList, mockGroups, "")
+	for priority, tasksInPriority := range scheduledTasksByPriority {
+		switch priority {
+		case lowerPriority:
+			expectedTasks := make([]*mockTask, 0)
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		case higherPriority:
+			expectedTasks := []*mockTask{tasks[4]}
+			assertEqualToAllocate(t, tasksInPriority, expectedTasks)
+		default:
+			panic("unexpected priority")
+		}
+	}
+}
+
+func TestPrioritySchedulingNoPreemption(t *testing.T) {
+	lowerPriority := 50
+	higherPriority := 40
+
+	agents := []*mockAgent{
+		{id: "agent1", slots: 4},
+		{id: "agent2", slots: 4},
+	}
+	groups := []*mockGroup{
+		{id: "group1", priority: &lowerPriority},
+		{id: "group2", priority: &higherPriority},
+	}
+	tasks := []*mockTask{
+		{id: "task1", slotsNeeded: 4, group: groups[0]},
+		{id: "task2", slotsNeeded: 1, group: groups[0]},
+		{id: "task3", slotsNeeded: 1, group: groups[1]},
+		{id: "task4", slotsNeeded: 0, group: groups[1]},
+		{id: "task5", slotsNeeded: 4, group: groups[1]},
+		{id: "task6", slotsNeeded: 0, group: groups[0]},
+	}
+
+	system := actor.NewSystem(t.Name())
+	taskList, groupMap, agentMap := setupSchedulerStates(t, system, tasks, groups, agents)
+
+	p := &priorityScheduler{}
+	toAllocate, _ := p.prioritySchedule(taskList, groupMap, agentMap, BestFit)
+
+	expectedToAllocate := []*mockTask{tasks[1], tasks[2], tasks[3], tasks[4], tasks[5]}
+	assertEqualToAllocate(t, toAllocate, expectedToAllocate)
+
+	// Check that agent stat has not changed.
+	for _, agent := range agentMap {
+		assert.Equal(t, agent.numEmptySlots(), 4)
+	}
+}
+
+func TestPrioritySchedulingNoPreemptionCase2(t *testing.T) {
```

Comment:
blocking: this function name should be more descriptive on what to test.

Reply:
Do you have a suggestion for a better name?

Reply:
I was confused by "NoPreemption". Maybe just `TestPrioritySchedulingBasic2`.

Reply:
updated to: `TestPrioritySchedulingPreemptionDisabledHigherPriorityBlocksLowerPriority`